test/test_messaging.py

```diff
@@ -210,7 +210,11 @@ def method(self, payload):
             backoff_queue = rabbit_manager.get_queue(
                 vhost, get_backoff_queue_name(delay)
             )
-            assert backoff_queue['messages'] == 0
+            try:
+                assert backoff_queue['messages'] == 0
+            except KeyError:
+                # When no messages in queue, the key may not be in the dict.
+                pass
```

Comment:
This looks like it is induced by a race condition.  I am not sure how I could make this test less brittle.

Reply:
I've never seen these tests fail before, locally or on Travis. I guess they fail on your local box?

The grace period before an expired queue is removed is controlled by the `EXPIRY_GRACE_PERIOD` constant, which is set to 5 seconds. If the test is brittle because it races against the queues being removed, lengthening that grace period should fix it, but it would be surprising if the test is taking longer than that to execute.

Reply:
I've merged  `-Xtheirs upstream upgrade-kombu` and I still got the issue.  For the record, I am running RabbitMQ 3.7.8 Erlang 21.1 installed using the helm chart from a Minikube Kubernetes cluster.  Tox is running a this very moment and I'll investigate that grace period constant

Reply:
I've doubled the grace period and clearly the tests do not take longer than that:  The queue still exists.

The HTTP API simply will not provide the `messages` key in its JSON payload.  Why and when this happens, I don't know.